src/poly/domain.rs

```diff
@@ -376,4 +379,20 @@ impl<G: Group> EvaluationDomain<G> {
     pub fn get_quotient_poly_degree(&self) -> usize {
         self.quotient_poly_degree as usize
     }
+
+    /// Hashes the constants in the domain which influence the proof into a u64
+    pub fn hash(&self) -> u64 {
```

Comment:
This isn't going to work; the `DefaultHasher` in the standard library is not guaranteed to have a stable algorithm. You need to take in a mutable reference to a BLAKE2b hasher as input and update it with the constants.

Also, we can't hash things down into `u64`s because it would lead to trivial collisions.